api/server.go

```diff
@@ -422,20 +423,29 @@ func (s *server) txpoolBroadcastHandler(jc jape.Context) {
 		}
 	}
 	if len(tbr.V2Transactions) != 0 {
+		var err error
+		// Overwrites the proofs for siacoin elements that are tracked in the database. Makes it slightly
+		// more convenient and less error-prone for users to broadcast v2 transactions since the correct
+		// proof can be filled implicitly. Unfortunately, that hides bad implementations from the
+		// implementor. In practice, this trade off is worth it.
+		// In full index mode, any UTXO can have its proofs overwritten.
+		// In personal index mode, only UTXOs that are registered to a wallet can have its proofs overwritten.
+		tbr.Basis, tbr.V2Transactions, err = s.wm.OverwriteElementProofs(tbr.V2Transactions)
+		if jc.Check("couldn't overwrite proofs", err) != nil {
```

Comment:
This is only safe when there are no v1 txns right? Also when we are in index mode.